nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/realm/Realm.java

```diff
@@ -60,108 +60,42 @@
 import org.jvnet.hk2.annotations.Contract;
 
 import com.sun.enterprise.security.SecurityLoggerInfo;
-import com.sun.enterprise.security.util.IASSecurityException;
 import com.sun.enterprise.util.LocalStringManagerImpl;
 
 /**
  *
- * @see java.security.Principal
+ * 
  * @author Harish Prabandham
  * @author Harpreet Singh
  * @author Jyri Virkki
  * @author Shing Wai Chan
  *
+ * @see java.security.Principal
  */
 @Contract
-public abstract class Realm implements Comparable<Realm> {
+public abstract class Realm extends BaseRealm implements Comparable<Realm> {
```

Comment:
I think this badly affects binary compatibility - hierarchy changed, static methods changed their containing classes. Realms written against previous API will fail to run after merging.

Reply:
I'm not keen on breaking binary compatibility for some slight refactoring

Reply:
It's not breaking binary compatibility is it? No static methods moved at all. Only the private state and instance methods have been moved, which are inherited by Realm.

Reply:
I've generated [revapi](https://revapi.org/) report against 5.191, it turns out, that it is ok. The only breaking change was caused by replacement of `javax.security.auth.x500.X500Principal` in method parameters, which was in different PR and is deprecated/for removal in JDK.

The diff was too big to apply in my head :)